build/tools/msvs/officialbuild.bat

```diff
@@ -28,66 +28,26 @@ if "%1" == "vc14x" (
   set compvers=vc14x
   call "%VS140COMNTOOLS%VsDevCmd.bat"
 )
-if "%1" == "vc120" (
-  @echo Building for vc120 / vs2013
-  set comp=120
-  set compvers=vc120
-  call "%VS120COMNTOOLS%VsDevCmd.bat"
-)
-if "%1" == "vc110" (
-  @echo Building for vc110 / vs2012
-  set comp=110
-  set compvers=vc110
-  call "%VS110COMNTOOLS%VsDevCmd.bat"
-)
-if "%1" == "vc100" (
-  @echo Building for vc100 / vs2010
-  set comp=100
-  set compvers=vc100
-  if "%WINDOWS71SDK%" == "" set WINDOWS71SDK=C:\Program Files\Microsoft SDKs\Windows\v7.1\Bin\
-  call "%WINDOWS71SDK%SetEnv.cmd"
-)
-if "%1" == "vc90" (
-  @echo Building for vc90 / vs2008
-  set comp=90
-  set compvers=vc90
-  if "%WINDOWS61SDK%" == "" set WINDOWS61SDK=C:\Program Files\Microsoft SDKs\Windows\v6.1\Bin\
-)
 
 if %compvers% == "Unknown" goto ERR_UNKNOWNCOMP
 
-if %compvers% == "vc90" (
-@echo ============================================================
-@echo This will only succeed if run from a SDK 6.1 command prompt.
-@echo ============================================================
-)
-
 rem Return to the build directory in case we have been moved elsewhere.
 
 cd %VSCMD_START_DIR%
 
 @echo Removing the existing destination so that a complete rebuild occurs.
 
-rmdir %compvers%_mswuddll /s /q
 rmdir %compvers%_mswuddll_x64 /s /q
-rmdir %compvers%_mswudll /s /q
 rmdir %compvers%_mswudll_x64 /s /q
 
-rmdir ..\..\lib\%compvers%_dll /s /q
 rmdir ..\..\lib\%compvers%_x64_dll /s /q
 
 @echo Delete the build output files from the last run.
 
-del %compvers%x86_Debug.txt
-del %compvers%x86_Release.txt
 del %compvers%x64_Debug.txt
 del %compvers%x64_Release.txt
 
-if "%compvers%" == "vc14x" call "%VS140COMNTOOLS%..\..\VC\vcvarsall.bat" x64
-if "%compvers%" == "vc120" call "%VS120COMNTOOLS%..\..\VC\vcvarsall.bat" x86_amd64
-if "%compvers%" == "vc110" call "%VS110COMNTOOLS%..\..\VC\vcvarsall.bat" x86_amd64
-if "%compvers%" == "vc100" call "%WINDOWS71SDK%SetEnv.Cmd" /X64 /Release
-if "%compvers%" == "vc90"  call "%WINDOWS61SDK%SetEnv.Cmd" /X64 /Release
+if "%compvers%" == "vc14x" call "%VS140COMNTOOLS%..\..\VC\vcvarsall.bat" x64 10.0.17763.0
```

Comment:
@doscott @vadz 
Not sure if hardcoding the winsdk_version here is the best. Perhaps we could add it as an optional argument to `officialbuild.bat`, which could be passed to `vcvarsall.bat`?

Additionally, we can check currently used SDK version inside `officialbuild.bat`, after calling  `vcvarsall.bat` and before running nmake, looking at the system environment variable `WindowsSDKVersion` and warn when the version is newer than the last known working one (not sure if possible within a batch file)...

Of course, I am not saying to do that for 3.3.0 release, just a suggestion for future consideration. I could try doing it.

Reply:
Yes, hardcoding the SDK version is not great, but as these scripts are only used by Danny currently, I'm not too bothered by it. Your proposal looks good, however!

Reply:
It could be passed in as an argument from buildall.bat. I don't have any time right now and I wouldn't object to someone else taking over the roll of making the official binaries. I will try to find time in the next few days to see if the new Windows 10 SDK works ok. It will also be interesting to see if a binary produced with the current SDK plays well when used with a VS2022 application built with the latest Windows 11 SDK that caused problems with VS2015.